flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/FlinkKafkaConsumerBase.java

```diff
@@ -770,8 +777,8 @@ public final void notifyCheckpointComplete(long checkpointId) throws Exception {
 			return;
 		}
 
-		final AbstractFetcher<?, ?> fetcher = this.kafkaFetcher;
-		if (fetcher == null) {
+		final KafkaOffsetCommitter offsetCommitter = this.kafkaOffsetCommitter;
```

Comment:
why do we need this `final` local variable? `kafkaOffsetCommitter` seems to be "write-once" variable.